frontend/src/components/containers/ContainerHierarchy.js

```diff
@@ -102,31 +127,43 @@ const buildContainerTreeFromPath = (context, path) => {
   if (samples.length) {
     if (!isExploded) {
       children.push({
-        title: <span style={entryStyle}>
+        title: renderEntry(
           <Text type="secondary">
             {samples.length} sample{samples.length === 1 ? '' : 's'}{' '}
             (click to expand)
           </Text>
-        </span>,
+        ),
         icon: <EllipsisOutlined />,
         key: `${container.id}$samples`,
       })
     }
+    else if (isFetching) {
+      children.push({
+        title: renderEntry(
+          <Text type="secondary">
+            {samples.length} sample{samples.length === 1 ? '' : 's'}{' '}
+            (loading)
+          </Text>
+        ),
+        icon: <LoadingOutlined />,
+        key: `${container.id}$samples`,
+      })
+    }
     else {
       children.push(...samples.map(sampleId => {
         const sample = context.samplesByID[sampleId];
-        if (!sample)
+        if (!sample || sample.isFetching)
           return loadingEntry(sampleId);
         const sampleKind = context.sampleKinds.itemsByID[sample.sample_kind]?.name
         return {
-          title: <span style={entryStyle}>
+          title: renderEntry(
             <Link to={`/samples/${sample.id}`}>
               <strong>{sample.name}</strong>{' '}
               sample ({sampleKind})
             </Link>
-          </span>,
+          ),
           icon: <CheckOutlined />,
-          key: sampleId,
+          key: `sample-${sampleId}`,
           type: 'sample',
```

Comment:
Maybe this could be the PR to address the ctrl-click problem in the tree. I do not know if it's even possible, but at the moment we have the same information duplicated on the container page for the sole purpose of being able to ctrl-click links. The sample(s) listed in content are also listed in the tree below. I think we could get rid of the content section if we could allow ctrl-click to work in the tree.

Reply:
Also, i think the samples listed in the tree should have their coord listed. This may require a lot of loading but i think this information is likely to be desirable.

Reply:
Totally agree with Ulysse on these points :P